src/services/accountService.ts

```diff
@@ -0,0 +1,53 @@
+import { Request } from 'express';
+import { UniqueConstraintError } from 'sequelize';
+
+import User from '../db/models/user';
+import { BaseRepository } from '../repositories/baseRepository';
+import { ResponseData } from '../types/common';
+
+export class AccountService {
+  constructor(private accountRepo: BaseRepository<User>) {}
+
+  async createAccountService(req: Request): Promise<ResponseData> {
+    try {
+      if (req.query.mobileAuth) {
+        if (!req.body.phoneNumber) {
+          return {
+            statusCode: 400,
+            error: 'Phone number is required for mobile login',
+          };
+        }
+        const result = await this.accountRepo.create(req.body, {});
+        return {
+          statusCode: 201,
+          result,
+        };
+      } else {
+        if (!req.body.email) {
+          return {
+            statusCode: 400,
+            error: 'email is required for social login',
+          };
+        }
+        const result = await this.accountRepo.create(req.body, {});
+        return {
+          statusCode: 201,
+          result,
+        };
+      }
+    } catch (error) {
+      if (error instanceof UniqueConstraintError) {
+        return {
+          statusCode: 303,
```

Comment:
Can you add a logger and add logs in case of error for debugging

Reply:
Please raise a new ticket to add logs for the entire project. Will address this later